### ISSUE TITLE
Bitter Crepe Rattlesnake - Auction parameters can still be changed during ongoing auction

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2023-09-nounsbuilder-judging#301]
Bitter Crepe Rattlesnake

medium

# Auction parameters can still be changed during ongoing auction

## Summary
In the previous NounsBuilder contest, a medium severity [finding](https://github.com/code-423n4/2022-09-nouns-builder-findings/issues/450) showed that auction parameters can be changed at anytime, even during ongoing auctions, and take effect immediately. 

The NounsBuilder team has chosen to rectify this by adding a `whenPaused` modifier, however this does little to no effect to mitigate the original issue.
## Vulnerability Detail
There are five setter functions:
- `setDuration`
- `setReservePrice`
- `setTimeBuffer`
- `setMinimumBidIncrement`
- `setFounderReward`

Each of these can be changed by the owner when the contract is `paused`. 

However, `pause()` and `unpause()` are also called by the owner. So the addition of the `whenPaused` modifier to the five setter functions does little to mitigate the impact. 
## Impact
Consider a scenario where there is an ongoing auction, owner calls `pause()`, changes timeBuffer through `setTimeBuffer()` and then calls `unpause()`. Bidders who previously had expectations based on the previous `timeBuffer` may expect their bid to extend an auction when it does not (based on the new timeBuffer) and are therefore negatively affected. 

The other functions can similar introduce unexpected changes and behaviours.

## Code Snippet
[https://github.com/sherlock-audit/2023-09-nounsbuilder/blob/main/nouns-protocol/src/auction/Auction.sol#L408](https://github.com/sherlock-audit/2023-09-nounsbuilder/blob/main/nouns-protocol/src/auction/Auction.sol#L408)

## Tool used
Manual Review

## Recommendation
As per previous recommendations, do not apply changes to ongoing auctions. Also, add a timelock for the changes.

**Comment by giraffe0x** [comment]
Request to escalate.

1) The fact that the protocol decided to fix this issue after it was raised in the previous C4 contest shows that this is unintended design. This report is arguing that the fix is ineffective. 

2) Judge may argue that the owner (DAO) is to be trusted and will not adjust auction parameters maliciously. True but even non-malicious adjustments can cause unintended consequences that might not be fair to previous bidders. [#271](https://github.com/sherlock-audit/2023-09-nounsbuilder-judging/issues/271) (duplicate) highlights this well. 

In summary, I think the protocol should come out and state whether it is by design that ongoing auction parameters can be changed -- which would then determine whether this report is valid.

**Comment by 0xnicchew** [comment]
Escalate

**Comment by sherlock-admin2** [comment]
> Escalate

    You've deleted an escalation for this issue.

**Comment by nevillehuang** [comment]
I think this detail in READ.ME is sufficient:

> Is the admin/owner of the protocol/contracts TRUSTED or RESTRICTED?

> TRUSTED

In this case, DAO is the trusted owner of the contracts, meaning they are responsible to consider the timing of parameters change during pause/unpause, so I don't have any further input, still suggest to maintain as invalid.

Also a proposal needs to be put up to change this parameters, and so it is highly unlikely that users will not notice during the voting phase.

**Comment by Oot2k** [comment]
Agree with @nevillehuang, this can be seen as admin issue and user input issue. If the user decides to bid on an auction that is clearly paused / ongoing proposal its his own fault. 
Change of prams is well known in advance, and auction can even be paused for the time of change.
Low/info
